src/main/java/frc/robot/subsystems/ball/intake/IntakeImpl.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.subsystems.ball.intake;
+
+import edu.wpi.first.wpilibj.Solenoid;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
+import java.util.EnumMap;
+import java.util.concurrent.TimeUnit;
+
+public class IntakeImpl extends RepeatingPooledSubsystem implements Intake {
+
+    public enum motorDirections {
+        FORWARDS,
+        BACKWARDS
+    }
+
+    EnumMap<motorDirections, Integer> motorDirectionsMap;
```

Comment:
You can actually skip this part altogether and make it way simpler. Enums can have their own fields, and you can add methods like getters to enums too, since Java enums are really class types, not just candied primitives (ints) like in a lot of other languages. It'll look something like this:

```java
enum MotorDirection {
  FORWARD(1),
  NONE(0),
  BACKWARD(-1),
  ;
  private final int multiplier;
  MotorDirection(int m) {
    this.multiplier = m;
  }
  public int getMultiplier() {
    return this.multiplier;
  }
}
```

Reply:
that would have to be outside of the class though, right? Because trying to do
```java
enum MotorDirection {
FORWARD(1),
etc...
```
it would call the constructor of the class with that as the parameter, right?

Reply:
No, it looks funny but that's how you set the fields of the enum members. That list of enum member values is basically a list of instances that will be instantiated, so you can treat them as if they're constructor invocations (they really are, under the hood).

Reply:
ex.: https://github.com/FRC3161/Yggdrasil/blob/master/src/main/java/ca/team3161/lib/utils/controls/LogitechDualAction.java#L47

Reply:
Since it's already implemented, is it even worth it to change it now though?

Reply:
Less bug-prone using the more conventional way I wrote it IMO, since it will fail to compile if you don't assign a direction to one of the enum members, but no big deal.